python/ffsim/variational/ucj_spin_balanced.py

```diff
@@ -429,11 +457,36 @@ def from_t_amplitudes(
 
         nocc, _, nvrt, _ = t2.shape
         norb = nocc + nvrt
+        if optimize:
+            diag_coulomb_indices = []
+            if pairs_aa is not None:
+                diag_coulomb_indices += pairs_aa
+            if pairs_ab is not None:
+                diag_coulomb_indices += pairs_ab
+            diag_coulomb_indices = list(set(diag_coulomb_indices))
```

Comment:
If `interaction_pairs` is None, this should end up being None. Instead, it ends up being the empty list.